service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -436,7 +439,8 @@ private void validateUniqueAiNotebookInstance(
             .addValue("workspace_id", notebookResource.getWorkspaceId().toString())
             .addValue("instance_id", notebookResource.getInstanceId())
             .addValue("location", notebookResource.getLocation());
-    Integer matchingCount = jdbcTemplate.queryForObject(sql, sqlParams, Integer.class);
+    Integer matchingCount =
+        DbRetryUtils.queryForObjectWithRetries(jdbcTemplate, sql, sqlParams, Integer.class);
```

Comment:
nit: you can make another overload like `queryForCountWithRetries` so you don't have to pass `Integer.class` in these places.

Reply:
I like having the `DbRetryUtils` methods wrap the jdbcTemplate methods closely here - I personally think the clarity from that outweighs the extra verbosity. It's a pretty weak preference, though